lib/sample_uploader/utils/sample_utils.py

```diff
@@ -327,6 +328,36 @@ def save_sample(sample, sample_url, token, previous_version=None):
     sample_ver = resp_json['result'][0]['version']
 
     print('saved sample {} (version: {}'.format(sample_id, sample_ver))
+
+    if previous_version and propagate_links:
+        print('start propagating previous data links')
+
+        data_links = get_data_links_from_sample(previous_version['id'],
+                                                previous_version['version'],
+                                                sample_url, token)
+
+        ss = SampleService(sample_url, token=token)
+
+        for data_link in data_links:
```

Comment:
I'm somewhat worried about the number of calls this will need to make... it also seems like this is something that might be better handled on the sampleservice side. I can imagine a situation where we modify a sample by directly calling sample_service in the future, such as if we build a sample editor.